src/Shared/Recruit.Vacancies.Client/Infrastructure/Client/QaVacancyClient.cs

```diff
@@ -72,14 +74,41 @@ public Task<Qualifications> GetCandidateQualificationsAsync()
 
         public async Task<QaDashboard> GetDashboardAsync()
         {
-            var dashboard = await _queryStoreReader.GetQaDashboardAsync();
+            var dashboard = await _queryStoreReader.GetQaDashboardAsync().ConfigureAwait(true);
 
             //todo - will be deleted
-            var allReviews = await _reviewRepository.GetActiveAsync();
+            var allReviews = await _reviewRepository.GetActiveAsync().ConfigureAwait(true);
             dashboard.AllReviews = allReviews.ToList();
+
             return dashboard;
         }
 
+        public async Task<List<VacancyReviewSearch>> GetSearchResultsAsync(string searchTerm)
+        {
+            if (TryGetVacancyReference(searchTerm, out var vacancyReference))
+            {
+                return await _reviewRepository.SearchAsync(vacancyReference).ConfigureAwait(false);
+            }
+            return new List<VacancyReviewSearch>();
+        }
+
+        /// <summary>
+        /// This could be somewhere more common, we can refactor when we come across a second use
+        /// </summary>
+        private static bool TryGetVacancyReference(string value, out long vacancyReference)
+        {
+            vacancyReference = 0;
+            if (string.IsNullOrEmpty(value)) return false;
+            
+            var regex = new Regex(@"^(VAC)?(\d{10})$", RegexOptions.IgnoreCase);
```

Comment:
maybe just check `value.Replace("vac", string.Empty).Replace("VAC", string.empty)` and then long try parse the result. More to avoid Regexing

Reply:
I think regex works better in this case as not only is checks for the string case insensitively, it also makes sure that the number has correct length.